core/src/test/java/com/xmartlabs/suspendedlocation/core/TestGeocode.kt

```diff
@@ -0,0 +1,163 @@
+package com.xmartlabs.suspendedlocation.core
+
+import android.content.Context
+import android.location.Address
+import android.location.Geocoder
+import com.google.android.gms.location.FusedLocationProviderClient
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockkClass
+import io.mockk.mockkConstructor
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+
+class TestGeocode {
+  private val testDispatcher = Dispatchers.IO
+
+  @MockK
+  lateinit var context: Context
+
+  private lateinit var suspendedLocationImpl: SuspendedLocationImpl
+
+  val address = mockkClass(Address::class)
+  val address1 = mockkClass(Address::class)
+  val address2 = mockkClass(Address::class)
+  val address3 = mockkClass(Address::class)
+  val address4 = mockkClass(Address::class)
+  val expectedList = listOf(
+      address,
+      address1,
+      address2,
+      address3,
+      address4
+  )
+
+  val locationName = "Xmartlabs"
+
+  @Before
+  fun setup() {
+    MockKAnnotations.init(this, relaxUnitFun = true)
+
+    every {
+      address.featureName = "Something"
+    }
+    val locationServices = mockkClass(FusedLocationProviderClient::class)
+    mockkConstructor(Geocoder::class)
+    every { anyConstructed<Geocoder>().getFromLocationName(locationName, 5) } returns expectedList
+
+    every { anyConstructed<Geocoder>().getFromLocationName(locationName, 1) } returns listOf(
+        address
```

Comment:
5 and 1 shouldn't be mocked in the setup, it could change. You have two options, use constants or mock in each test